IMPLEMENTATION.md

```diff
@@ -0,0 +1,98 @@
+ 
+- Work in progress: 
+  - `./src/implicitTiling`: Classes that support accessing implicit tiling information
+  - `./src/traversal`: Classes for traversing tilesets
+  - `./src/metadata`: Classes that support accessing 3D Metadata
+
+- `./specs`: Jasmine spec drafts
+- `./specs/data`: Test data
+
+## Overview
+
+The classes in the `./src/validation` directory are the core classes for the 3D Tiles validation. 
+
+The entry point for the tileset validation is the `TilesetValidator` class:
+
+- It receives a tileset JSON string and parses it into a `structure/Tileset.ts` object.
+  - Note: These `structure/*` classes are "plain objects". The do not have methods, and no real type checking. They _only_ hold the parsed data.
+- It traverses the `Tileset` structure, and performs the validation
+  - For each object type, there is a validator, like `AssetValidator`, `ContentValidator`, `TileValidator`, `BoundingVolumeValidator` etc. For now, these are mainly classes with single static functions, but they might be refined for some objects.
+
+Each validation function receives a `ValidationContext` as the last parameter. 
+
+The `ValidationContext` class:
+
+- Can contain configuration options that may be needed in the future (like `doValidateContents=false` or so)
+- Maintains a `ResourceResolver` that reads a `Buffer` from a given URI (resolved in the respective context - the tileset, external tileset, or glTF)
+- Collects the `ValidationIssue` instances (errors and warnings) that are found during the traversal
+
+The `ValidationIssue` class and its types:
+
+- The `ValidationIssue` itself is a plain data structure with basic info about the issue (message, path, severity...)
+- Instances of this class are created with methods in the `issues*` classes, roughly categorized here:
+  - `issues/IoValidationIssues.ts`: Fundamental IO errors, like `JSON_PARSE_ERROR`
+  - `issues/JsonValidationIssues.ts`: Violations of the JSON schema, like `PROPERTY_MISSING` or `ARRAY_LENGTH_MISMATCH`
+  - `issues/StructureValidationIssues.ts`: General issues related to an inconsistent structure, like `IDENTIFIER_NOT_FOUND` 
+  - `issues/SemanticValidationIssues.ts`: Issues related to inconsistent property values, like `TILE_GEOMETRIC_ERROR_INCONSISTENT` 
+  - `ContentValidationIssues.ts`: An error or warning in a tile content
+- For validation issues that refer to the tile content, each `ValidationIssue` can have an array of `internalIssues`. This can be filled, for example, with the information from the glTF validator that caused the validation to fail
+
+## Discussion points
+
+### General functionality
+
+- It should be possible to "batch process" a directory (with caveats - how to detect whether a file is a `tileset.json`, beyond its name?)
+- It should be possible to write reports into files.
```

Comment:
+1 for both of these.

> how to detect whether a file is a `tileset.json`, beyond its name?

I think just checking the name is ok.

Reply:
Using a plain name check _might_ be too narrow. Nobody knows what people might call their tilesets. The extreme case would be to allow the user (via config options) to define a regex for the filenames to be processed, with `tileset.json` being the default, or things like `tileset[.*].json` for things like `tileset_v1.json`.

But I consider this point to be on the radar due to the note in the IMPLEMENTATION.md